make test fails with latest code

From @MichaelLarkin and @gafferongames 

```
Glenn Fiedler:spiral_calendar_pad:  9:14 PM
Hey folks, might be a heisenbug in tests.
9:14
Not 100% sure, but Michael is getting this with “make test” with latest backend repo code
9:14
--- FAIL: TestRelayInitAddressIsInvalid (0.02s)
  relay_init_handler_test.go:124: 
    Error Trace: relay_init_handler_test.go:124
          relay_init_handler_test.go:340
    Error:   Not equal: 
          expected: 400
          actual : 200
    Test:   TestRelayInitAddressIsInvalid
--- FAIL: TestRelayUpdateInvalidAddress (0.02s)
  relay_update_handler_test.go:70: 
    Error Trace: relay_update_handler_test.go:70
          relay_update_handler_test.go:207
    Error:   Not equal: 
          expected: 400
          actual : 404
    Test:   TestRelayUpdateInvalidAddress
FAIL
coverage: 85.2% of statements
FAIL github.com/networknext/backend/transport 0.498s
FAIL
make: *** [test-unit-backend] Error 1





9:14
I’m not able to repro, and we’re both running MacOS. I’ve checked his go version and everything looks legit.
9:15
Only thing I have left is, isn’t this one of the tests you were playing around with today @Jonathan? eg. JSON vs. Binary handling.
9:15
That’s all I got.
9:15
Help? :smile:
```

--- a/transport/relay_init_handler_test.go
+++ b/transport/relay_init_handler_test.go
@@ -6,13 +6,13 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"errors"
-	"fmt"
 	"math"
 	mrand "math/rand"
 	"net"
 	"net/http"
 	"net/http/httptest"
 	"os"
+	"strings"
 	"testing"
 	"time"
 
@@ -335,7 +335,9 @@ func TestRelayInitAddressIsInvalid(t *testing.T) {
 	{
 		buff, err := packet.MarshalJSON()
 		assert.NoError(t, err)
-		offset := len(fmt.Sprintf("{\"encrypted_token\":\"%s\",\"magic_request_protection\":%d,\"nonce\":\"%s\",\"relay_address\":\"", base64.StdEncoding.EncodeToString(packet.EncryptedToken), packet.Magic, base64.StdEncoding.EncodeToString(packet.Nonce)))
+
+		offset := strings.Index(string(buff), "127.0.0.1:40000")
+		assert.GreaterOrEqual(t, offset, 0)
 		buff[offset] = 'x' // first number in ip address is now 'x'
 		relayInitAssertions(t, "application/json", relay, buff, http.StatusBadRequest, nil, nil, nil, nil, routerPrivateKey[:])
 	}

